pkg/synchronizer/kubernetes/synchronizer.go

```diff
@@ -305,7 +305,9 @@ func (sync *KubeSynchronizer) ProcessSubResources(appsub *appv1alpha1.Subscripti
 
 	err := sync.SyncAppsubClusterStatus(appsub, appsubClusterStatus, nil, nil)
 	if err != nil {
-		klog.Warning("error while sync app sub cluster status: ", err)
+               klog.Error("error while sync app sub cluster status: ", err)
+               sync.kmtx.Unlock()
```

Comment:
Maybe the `sync.kmtx.Unlock()` could be removed, after adding a return. https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/adfb7f230f2dbcc826a5c0832a762386a9ea6500/pkg/synchronizer/kubernetes/synchronizer.go#L127